typescript/cli/src/config/chain.ts

```diff
@@ -90,6 +90,62 @@ export async function createChainConfig({
     protocol: ProtocolType.Ethereum,
     rpcUrls: [{ http: rpcUrl }],
   };
+  const wantAdvancedConfig = await confirm({
+    message: 'Do you want to add advanced config for this chain?',
+  });
+  if (wantAdvancedConfig) {
+    const wantBlockConfig = await confirm({
+      message: 'Do you want to add block config for this chain?',
+    });
+    if (wantBlockConfig) {
+      const blockConfirmation = await input({
+        message:
+          'Enter no. of blocks to wait before considering a transaction confirmed(0-10)',
+        validate: (value) => parseInt(value) >= 0 && parseInt(value) <= 10,
+      });
+      const blockReorgPeriod = await input({
+        message:
+          'Enter no. of blocks before a transaction has a near-zero chance of reverting(0-50)',
+        validate: (value) => parseInt(value) >= 0 && parseInt(value) <= 50,
+      });
+      const blockTimeEstimate = await input({
+        message: 'Enter the rough estimate of time per block in seconds(0-20)',
+        validate: (value) => parseInt(value) >= 0 && parseInt(value) <= 20,
+      });
+      metadata.blocks = {
+        confirmations: parseInt(blockConfirmation, 10),
+        reorgPeriod: parseInt(blockReorgPeriod, 10),
+        estimateBlockTime: parseInt(blockTimeEstimate, 10),
+      };
+    }
+    const wantGasConfig = await confirm({
+      message: 'Do you want to add gas config for this chain?',
+    });
+    if (wantGasConfig) {
+      const isEIP1559 = await confirm({
+        message: 'Is your chain an EIP1559 enabled?',
+      });
+      if (isEIP1559) {
+        const maxFeePerGas = await input({
+          message: 'Enter the max fee per gas in gwei',
+        });
+        const maxPriorityFeePerGas = await input({
+          message: 'Enter the max priority fee per gas in gwei',
+        });
+        metadata.transactionOverrides = {
+          maxFeePerGas: parseInt(maxFeePerGas, 10) * 10 ** 9,
+          maxPriorityFeePerGas: parseInt(maxPriorityFeePerGas, 10) * 10 ** 9,
```

Comment:
Why `** 9` here? Did you mean to use 18 for Ether's 18 decimals?
Either way, there's a possibility of Number overrun so you should use `bigint` instead.

Reply:
That **9 is to refactor it in terms of gwei. I will update it with bigint